lib/internal/fs/dir.js

```diff
@@ -39,9 +40,19 @@ class Dir {
     this[kDirPath] = path;
     this[kDirClosed] = false;
 
-    this[kDirOptions] = getOptions(options, {
-      encoding: 'utf8'
-    });
+    this[kDirOptions] = {
+      bufferSize: 32,
+      ...getOptions(options, {
+        encoding: 'utf8'
+      })
+    };
+
+    if (typeof this[kDirOptions].bufferSize !== 'number' ||
```

Comment:
Optional nit, but this logic pretty much duplicates the `validateInteger()` function in `lib/internal/validators.js` if you wanted to reuse that.

Reply:
@cjihrig Yeah, I’ve made this use `validateUint32()` and adjusted the tests to match against the other error types